subprojects/base-services/src/main/java/org/gradle/internal/hash/FileHasher.java

```diff
@@ -24,7 +24,7 @@
 
 public interface FileHasher {
     /**
-     * Returns the hash of the given input stream.
+     * Returns the hash of the given input stream. This method buffers reads on the stream.
```

Comment:
We should add this in several other places, too. Or better: review how we use buffers in general, because we never paid a lot of attention to that.

We've managed to get some significant improvements in task output packaging by fine-tuning buffers:

![](https://user-images.githubusercontent.com/495366/28706148-3fc2cc2a-7373-11e7-8236-b49c47e246cb.png)

This also shows how using a(nother) buffer can sometimes reduce performance instead of increasing it.